### PR TITLE
REPO-3876: Fix Remote User Mapper

### PR DESCRIPTION
**Pull request by eknizat**
* Set timeout to the http client used by keycloak to retrieve a public key
* Ensure that the user mapper does not throw exceptions
* IdentityServiceRemoteUserMapper should not attempt basic authentication

### CODE REVIEW

File: src/main/java/org/alfresco/repo/security/authentication/identityservice/IdentityServiceRemoteUserMapper.java [hunk 1]
```diff
@@ -179,28 +187,10 @@ private String extractUserFromHeader(HttpServletRequest request)
         }
         else
         {
-            // if bearer token failed, try basic auth, if enabled
-            if (this.keycloakDeployment.isEnableBasicAuth())
+            // Basic auth in IdentityServiceRemoteUserMapper was removed as part of https://issues.alfresco.com/jira/browse/REPO-3876
```
Comment (skopf) [inline src/main/java/org/alfresco/repo/security/authentication/identityservice/IdentityServiceRemoteUserMapper.java]: Not sure if we need this comment

File: src/main/java/org/alfresco/repo/security/authentication/identityservice/IdentityServiceDeploymentFactoryBean.java [hunk 2]
```diff
@@ -51,7 +55,14 @@ public void setIdentityServiceConfig(IdentityServiceConfig config)
     public KeycloakDeployment getObject() throws Exception
     {
         KeycloakDeployment deployment = KeycloakDeploymentBuilder.build(this.identityServiceConfig);
-        
+
+        // Set client with custom timeout values. This can be removed if the future versions of Keycloak accept timeout values through the config.
+        HttpClient client = new HttpClientBuilder()
+                .establishConnectionTimeout(2, TimeUnit.SECONDS)
+                .socketTimeout(2, TimeUnit.SECONDS)
```
Comment (skopf) [inline src/main/java/org/alfresco/repo/security/authentication/identityservice/IdentityServiceDeploymentFactoryBean.java]: 1. Are there other timeouts as well? What if we could establish the connection real quick and the TCP connection is stable, but the other end simply does not respond? There should be another timeout on the HTTP level
2. Not sure if there are situations where we need to increase this timeout. perhaps we should read these values from prop files in the end. Just to have a possibility for support to increase them in some edge case scenarios

File: src/test/resources/alfresco-global.properties [hunk 3]
```diff
@@ -43,4 +43,6 @@ identity-service.enable-pkce=true
 identity-service.ignore-oauth-query-parameter=true
 identity-service.credentials.secret=11111
 identity-service.credentials.provider=secret
+identity-service.client-socket-timeout=1000
```
Comment (killerboot) [inline src/test/resources/alfresco-global.properties]: Please add a comment about the units, milliseconds in this case.

File: src/test/java/org/alfresco/repo/security/authentication/identityservice/IdentityServiceRemoteUserMapperTest.java [hunk 4]
```diff
@@ -85,24 +85,24 @@
...
-    private IdentityServiceConfig identityServiceConfig;
-
-    /* (non-Javadoc)
-     * @see junit.framework.TestCase#setUp()
-     */
-    @Override
-    protected void setUp() throws Exception
+    private IdentityServiceConfig identityServiceConfig;
+
+    /* (non-Javadoc)
```
Comment (killerboot) [inline src/test/java/org/alfresco/repo/security/authentication/identityservice/IdentityServiceRemoteUserMapperTest.java]: We don't need such JavaDocs any more, do we?

File: src/test/java/org/alfresco/repo/security/authentication/identityservice/IdentityServiceRemoteUserMapperTest.java [hunk 5]
```diff
@@ -113,19 +113,19 @@ protected void setUp() throws Exception
...
-    protected void tearDown() throws Exception
-    {
-        childApplicationContextManager.destroy();
-        childApplicationContextManager = null;
-        childApplicationContextFactory = null;
-    }
+                    getApplicationContext().getBean(CONFIG_BEAN_NAME);
+    }
+    
+    /* (non-Javadoc)
```
Comment (killerboot) [inline src/test/java/org/alfresco/repo/security/authentication/identityservice/IdentityServiceRemoteUserMapperTest.java]: Odd JavaDoc...
